### PR TITLE
q4 and q5

### PR DESCRIPTION
**Pull request by ziggybaz**
@rebeccawaweru 
Completed the pending files, review

### CODE REVIEW

**Comment by ziggybaz** [comment]
Just added unit test coverage for file storage

**Comment by ziggybaz** [comment]
Added q8
The failing check is due to the subsequent problem not yet being solved

**Comment by rebeccawaweru** [comment]
You have a lot of repetition in base_model.py since you have also added code for questions I had already done. Please remove all the repetitions and incooporate only the code that answers your assigned questions.

**Comment by rebeccawaweru** [comment]
You have not done the unittest for question 4....

**Comment by rebeccawaweru** [comment]
The unittest for question 4 should have been in the same file as the unittest for question 3,,,,you've not pushed any changes for the test_base_model.py

**Comment by ziggybaz** [comment]
> You have a lot of repetition in base_model.py since you have also added code for questions I had already done. Please remove all the repetitions and incooporate only the code that answers your assigned questions.

done

**Comment by ziggybaz** [comment]
> The unittest for question 4 should have been in the same file as the unittest for question 3,,,,you've not pushed any changes for the test_base_model.py

exactly, you created the functions so you should have tested them, I've only edited one function based on the requirements of q4

**Comment by ziggybaz** [comment]
Also, when performing a review you go to the 'file changed' tab and write your comments there, this space is for conversations.

**Comment by ziggybaz** [comment]
Anyway, I've implemented the requested changes,

**Comment by ziggybaz** [comment]
Added q9, sorting out its failed workflow checks

**Comment by ziggybaz** [comment]
> Added q9, sorting out its failed workflow checks

Finished, now its requirement compliant except for the one unittest you were to fix

File: models/base_model.py
Comment (rebeccawaweru) [inline models/base_model.py:1]: I meant you need to add unittest for the args and kwargs in the test_base_model.....the existing tests i added did not include for question 4.

Comment (rebeccawaweru) [reply models/base_model.py:1]: You need to add test cases for the kwargs and args inside the existing class TestBaseModel_instantiation(unittest.TestCase):

Comment (ziggybaz) [reply models/base_model.py:1]: sorry, was really down today

**Pull request merged**